Modules/Dashboard/Models/Report.php

```
@@ -0,0 +1,706 @@
+    }
+
+    /**
+     * Format data as HTML
+     */
+    protected function formatAsHtml(array $data): string
+    {
+        $html = "<html><head><title>{$this->name}</title></head><body>";
+        $html .= "<h1>{$this->name}</h1>";
+        $html .= "<p>Generated: " . now()->format('Y-m-d H:i:s') . "</p>";
+
+        foreach ($data['data'] as $sourceName => $sourceData) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$sourceData'. (Unused Code Rules)

(UnusedLocalVariable)